.ci/Jenkinsfile

```diff
@@ -314,43 +315,105 @@ def generateFunctionalTestStep(Map args = [:]){
+  }
+
   // Skip scenarios per platform
   tags += " && ~@skip:${goArch}"
 
-  return {
-    withNode(labels: "${workerLabels}", sleepMax: 20, forceWorkspace: true){
-      deleteDir()
-      unstash 'source'
-      withGoEnv(version: "${GO_VERSION}"){
-        try {
-          if(isInstalled(tool: 'docker', flag: '--version')) {
-            dockerLogin(secret: "${DOCKER_ELASTIC_SECRET}", registry: "${DOCKER_REGISTRY}")
-          }
-          filebeat(image: 'docker.elastic.co/beats/filebeat:7.13.0-SNAPSHOT', output: "docker_logs_${goArch}_${suite}_${name}.log", workdir: "${env.WORKSPACE}"){
-            dir("${BASE_DIR}"){
-              withEnv(["GOARCH=${goArch}", "ELASTIC_APM_GLOBAL_LABELS=build_pr=${isPR()},build_id=${env.BUILD_ID},go_arch=${goArch},beat_version=${env.BEAT_VERSION},stack_version=${env.STACK_VERSION}"]) {
-                withSecretVault(secret: "${VAULT_INSTRUMENTATION_SECRET}", user_key: "apmServerToken", user_var_name: "APM_SECRET_TOKEN", pass_key: "apmServerUrl", pass_var_name: "APM_SERVER_URL"){
-                  sh script: """.ci/scripts/functional-test.sh "${suite}" "${tags}" "${STACK_VERSION}" "${BEAT_VERSION}" """, label: "Run functional tests for ${suite}:${tags}"
+    // Setup environment for platform
+    def envContext = []
+    if (platform.contains("windows")) {
+        def mingwArch = is32() ? '32' : '64'
+        def chocoPath = 'C:\\ProgramData\\chocolatey\\bin'
+        def chocoPython3Path = 'C:\\Python38;C:\\Python38\\Scripts'
+        path = "${env.WORKSPACE}\\bin;${chocoPath};${chocoPython3Path};C:\\tools\\mingw${mingwArch}\\bin;C:\\windows\\system32;C:\\windows\\system32\\WindowsPowerShell\\v1.0\\"
+        linuxIp = grabWorkerIP()
+        envContext = [
+            "KIBANA_URL=https://${linuxIp}:5601",
+            "ELASTICSEARCH_URL=https://${linuxIp}:9200",
+            "FLEET_URL=https://${linuxIp}:8220",
+            "PROVIDER=remote",
+            "PATH=${path}",
+        ]
+    }
+    envContext.add("GOARCH=${goArch}")
+    envContext.add("ELASTIC_APM_GLOBAL_LABELS=build_pr=${isPR()},build_id=${env.BUILD_ID},go_arch=${goArch},beat_version=${env.BEAT_VERSION},stack_version=${env.STACK_VERSION}")
+
+    return {
+        withNode(labels: "${workerLabels}", sleepMax: 20, forceWorkspace: true){
+            deleteDir()
+            unstash 'source'
+            try {
+                if(isInstalled(tool: 'docker', flag: '--version')) {
+                    dockerLogin(secret: "${DOCKER_ELASTIC_SECRET}", registry: "${DOCKER_REGISTRY}")
                 }
-              }
+                dir("${BASE_DIR}") {
+                    // Doing a windows agent test, start up docker before the new environment is applied for windows
+                    if (platform.contains("windows")) {
+                        sh script: "docker-compose -f cli/config/compose/profiles/fleet/docker-compose.yml up -d", label: "Deploying fleet via docker"
+                        linuxIp = grabWorkerIP()
```

Comment:
What's the reason for this second call to the `grabWorkerIP`? Can it be removed and use the first call in https://github.com/elastic/e2e-testing/pull/1330/files#diff-ef8b2f5e4d21d81c0eba4754dea3a3dfc2c67d26284f195165b650cf4aff7beaR340


```suggestion
```

Reply:
Thanks this was missed when I was fixing a scoping issue